app/models/service_template_ansible_playbook.rb

```diff
@@ -44,8 +44,9 @@ def self.create_catalog_item(options, auth_user)
           next unless dialog_name
 
           job_template = enhanced_config.fetch_path(action, :configuration_template)
-          enhanced_config[action][:dialog] =
-            Dialog::AnsiblePlaybookServiceDialog.create_dialog(dialog_name, job_template)
+          new_dialog = Dialog::AnsiblePlaybookServiceDialog.create_dialog(dialog_name, job_template)
+          enhanced_config[action][:dialog] = new_dialog
+          service_template.options[:config_info][action][:dialog_id] = new_dialog.id
         end
```

Comment:
Does it make sense to extract the logic from Line 47 - 49 into a separate method? I think it may make what is happening in the `transaction` easier to read.